src/foam/nanos/auth/HumanNameTrait.js

```diff
@@ -32,12 +36,33 @@ foam.INTERFACE({
       documentation: 'Full legal name of user. Appends first, middle & last name.',
       transient: true,
       expression: function(firstName, middleName, lastName) {
-        return middleName != '' ? firstName + ' ' + middleName + ' ' + lastName : firstName + ' ' + lastName;
+        if ( firstName && middleName && lastName ) {
+          return `${firstName} ${middleName} ${lastName}`;
+        }
+
+        if ( firstName && lastName ) {
+          return `${firstName} ${lastName}`;
+        }
+
+        return '';
       },
       javaGetter: `
-        return ! getMiddleName().equals("")
-          ? getFirstName() + " " + getMiddleName() + " " + getLastName()
-          : getFirstName() + " " + getLastName();
+        String firstName = this.getFirstName();
+        String middleName = this.getMiddleName();
+        String lastName = this.getLastName();
+
+        if ( ! SafetyUtil.isEmpty(firstName) && 
+             ! SafetyUtil.isEmpty(middleName) &&
+             ! SafetyUtil.isEmpty(lastName) ) {
+
+          return firstName + " " + middleName + " " + lastName;
+        }
+
+        if ( ! SafetyUtil.isEmpty(firstName) && ! SafetyUtil.isEmpty(lastName) ) {
+          return firstName + " " + lastName;
+        }
+
```

Comment:
Do we want to support just 'lastName'?

Reply:
@kgrgreer documentation says full legal name so I returned if both first name and last name existed.

Reply:
I would think that some name is better than no name. If it's invalid to not specify both first and last name, then that should be a validation error.